kontonummer.js

```diff
@@ -20,378 +20,440 @@
-            lengths : {
-                clearing : 5,
-                account  : 10,
-                control  : 10
-            }
-        },{
-            name    : 'Ålandsbanken',
-            regex   : /^(23[0-9][0-9])([0-9]{7})$/,
-            modulo  : 11,
-            lengths : {
-                clearing : 4,
-                account  : 7,
-                control  : 11
-            }
-        }];
-    
+            return {
+                isValid: false,
+            };
+        }
+
         for (i in banks) {
             bank = banks[i];
-            ctrlNum = n.substr(-bank.lengths.control, bank.lengths.control);
-            if (bank.regex.test(n) && ((bank.modulo === 11 && mod11(ctrlNum)) || (bank.modulo === 10 && mod10(ctrlNum)))) {
-                return {
-                    bank_name       : bank.name,
-                    clearing_number : n.substr(0, bank.lengths.clearing),
-                    account_number  : n.substr(bank.lengths.clearing, bank.lengths.account)
-                };
+            var bankNumber = (bank.zerofill) ? fillZeros(n, bank) : n;
+
+            var numberChecksumValidation = validateChecksum(bank, bankNumber);
+            numberChecksumValidation.errors = numberChecksumValidation.errors.concat(validateLength(bank, bankNumber));
+            if (numberChecksumValidation.bankName) {
+                matchedBanks.push(numberChecksumValidation);
+
+                if (!numberChecksumValidation.errors.length) {
+                    isValid = true;
+                }
+            }
+        }
+
+
+        if (!matchedBanks.length) {
+            errors.push('unknown_clearing_number');
+        }
+
+        if (!isValid) {
+            return {
+                isValid: false,
+                errors: errors,
+                matchedBanks: matchedBanks
+            };
+        }
+
+        return {
+            isValid: isValid,
+            matchedBanks: matchedBanks
+        };
+    };
+
+    var validateLength = function(bank, bankNumber) {
+        var errors = [];
+
+        if (bank.regex.test(bankNumber)) {
+            if (bankNumber.length < bank.lengths.clearing + bank.lengths.account) {
+                errors.push('too_short');
+            } else if (bankNumber.length > bank.lengths.clearing + bank.lengths.account) {
+                errors.push('too_long');
+            }
+        }
+
+        return errors;
+    };
+
+    var validateChecksum = function(bank, bankNumber) {
+        var bankName = null;
+        var errors = [];
+        var ctrlNum = bankNumber.substr(-bank.lengths.control, bank.lengths.control);
+
+        if (bank.regex.test(bankNumber)) {
+            bankName = bank.name;
+            if (!(bank.modulo === 11 && mod11(ctrlNum)) && !(bank.modulo === 10 && mod10(ctrlNum))) {
+                errors.push('bad_checksum');
             }
         }
-        return false;
+
+        return {
+            errors: errors,
+            bankName: bankName,
```

Comment:
CamelCase is inconsistent with the other returned properties here and differs from the documented snake_case. On the other hand on error the top level returned properties like `isValid` and `matchedBanks` are CamelCase. Do we want to be consistent?

Reply:
I guess it's good to be consistent!

Reply:
Is it just the returned property we want turned into snake_case or all camelCased variables as well? Since the returned prop is documented as snake_case I get that one, but still not sure about the rest of the project as it's very mixed